processor/src/main/java/io/norberg/automatter/processor/FieldProcessor.java

```diff
@@ -0,0 +1,83 @@
+package io.norberg.automatter.processor;
+
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.FieldSpec;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.TypeName;
+
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public interface FieldProcessor {
+  FieldSpec builderField(Descriptor d, ExecutableElement field) throws AutoMatterProcessorException;
+  FieldSpec valueField(Descriptor d, ExecutableElement field) throws AutoMatterProcessorException;
+
+  Iterable<MethodSpec> accessors(Descriptor d, ExecutableElement field) throws AutoMatterProcessorException;
+
+  Iterable<Statement> defaultConstructor(Descriptor d, ExecutableElement field);
+  Iterable<Statement> copyValueConstructor(Descriptor d, ExecutableElement field) throws AutoMatterProcessorException;
+  Iterable<Statement> copyBuilderConstructor(Descriptor d, ExecutableElement field) throws AutoMatterProcessorException;
+  BuildStatements build(Descriptor d, ExecutableElement field) throws AutoMatterProcessorException;
+  Iterable<Statement> valueConstructor(Descriptor d, ExecutableElement field) throws AutoMatterProcessorException;
+
+  default TypeName fieldType(final Descriptor d, final ExecutableElement field) throws AutoMatterProcessorException {
```

Comment:
I guess these methods aren't actually part of the `FieldProcessor` interface. I.e., `FieldProcessor` users aren't expected to call them.

How about grouping and moving them to separate classes based on functionality? E.g. put the field introspection methods into e.g. a `Fields` class etc.